src/date-input/types.ts

```diff
@@ -9,55 +11,59 @@ export interface DateInputProps extends React.AriaAttributes {
 
     // Input-specific Attributes
     "data-testid"?: string | undefined;
+
+    /**
+     * The value of the date input in "YYYY-MM-DD" or "YYYY-M-D" string format.
+     * In the range variant, it serves as the start value
+     */
     value?: string | undefined;
-    endValue?: string | undefined;
+    /**
+     * The end value of the date input in "YYYY-MM-DD" or "YYYY-M-D"
+     * string format. Only relevant to range variant
+     */
+    valueEnd?: string | undefined;
     error?: boolean | undefined;
     disabled?: boolean | undefined;
     /** Dates to disable in `YYYY-MM-DD` format. Example: `["2023-04-30"]` */
     disabledDates?: string[] | undefined;
     /** The variant of the component. Values `single` | `range` */
     variant?: DateInputVariant | undefined;
-    /** Specifies if done/cancel buttons are visible in the calendar. Defaults to `true` */
+    /**
+     * Specifies if the "Done" and "Cancel" action buttons should be rendered.
+     * Is restricted to `true` on mobile viewports
+     */
     withButton?: boolean | undefined;
     /**
      * Restrict selection to within this date range, in `YYYY-MM-DD` format.
      * Example: `["2023-03-15", "2023-04-19"]
      */
     between?: [string, string] | undefined;
     /**
-     * Function that returns the values in the DateInput on valid selection in an object format
-     * as such { start: "YYYY-MM-DD", end: "YYYY-MM-DD"}
+     * Function that returns when a valid selection is made. Returns the start and
+     * end date in "YYYY-MM-DD" or "YYYY-M-D" string format.
      */
-    onChange?: ((value: ChangeValueTypes) => void) | undefined;
```

Comment:
onChange and onBlur wouldn't return YYYY-M-D string format.

Reply:
Yup thanks for pointing out. Mistake in documentation